apps/backend/api/services/Search/util.js

```diff
@@ -75,14 +75,17 @@ export const filterAndSortPosts = curry((opts, q) => {
     })
   }
 
-  if (afterTime) {
+  if (afterTime && beforeTime) {
```

Comment:
actually i just realized, might it not be good enough to just have the afterTime and beforeTime conditionals separate from each other? wouldnt that cover the case that both are set? also im trying to understand why the conditions in afterTime & beforeTime are different from the cases where only one is set. Seems like they should be the same, and then we only need to the two separate conditions

Reply:
i guess im not communicating this clearly enough. i still don't see why we need three conditions here: one for afterTime & beforeTime, one for just afterTime and one for just beforeTime. Can't we just have two separate if statements for afterTime and beforeTime, and that would handle it? like this:

if (afterTime) {
      q.where(q2 =>
        q2.where('posts.endTime', '>=', afterTime)
      )
  }

if (beforeTime) {
  q.where(q2 =>
      q2.where('posts.end_time', '<', beforeTime)
    )
}